modules/sdk-coin-eth/src/eth.ts

```diff
@@ -942,6 +1003,239 @@ export class Eth extends BaseCoin {
     }
   }
 
+  private signRecoveryTSS(
+    userKeyCombined: ECDSA.KeyCombined,
+    backupKeyCombined: ECDSA.KeyCombined,
+    txHex: string
+  ): ECDSAMethodTypes.Signature {
+    const MPC = new Ecdsa();
+
+    const signerOneIndex = userKeyCombined.xShare.i;
+    const signerTwoIndex = backupKeyCombined.xShare.i;
+    const signShares: ECDSA.SignShareRT = MPC.signShare(
+      userKeyCombined.xShare,
+      userKeyCombined.yShares[signerTwoIndex]
+    );
+
+    let signConvertS21: ECDSA.SignConvertRT = MPC.signConvert({
+      xShare: backupKeyCombined.xShare,
+      yShare: backupKeyCombined.yShares[signerOneIndex], // YShare corresponding to the other participant signerOne
+      kShare: signShares.kShare,
+    });
+
+    const signConvertS12: ECDSA.SignConvertRT = MPC.signConvert({
+      aShare: signConvertS21.aShare,
+      wShare: signShares.wShare,
+    });
+
+    signConvertS21 = MPC.signConvert({
+      muShare: signConvertS12.muShare,
+      bShare: signConvertS21.bShare,
+    });
+
+    const [signCombineOne, signCombineTwo] = [
+      MPC.signCombine({
+        gShare: signConvertS12.gShare as ECDSA.GShare,
+        signIndex: {
+          i: (signConvertS12.muShare as ECDSA.MUShare).i,
+          j: (signConvertS12.muShare as ECDSA.MUShare).j,
+        },
+      }),
+      MPC.signCombine({
+        gShare: signConvertS21.gShare as ECDSA.GShare,
+        signIndex: {
+          i: (signConvertS21.muShare as ECDSA.MUShare).i,
+          j: (signConvertS21.muShare as ECDSA.MUShare).j,
+        },
+      }),
+    ];
+
+    const MESSAGE = Buffer.from(txHex);
+
+    // const hashGenerator = (hashType?: string): Hash | undefined => {
+    //   return hashType === 'keccak256' ? createKeccakHash('keccak256') : undefined;
+    // };
+
+    const [signA, signB] = [
+      MPC.sign(MESSAGE, signCombineOne.oShare, signCombineTwo.dShare),
+      MPC.sign(MESSAGE, signCombineTwo.oShare, signCombineOne.dShare),
+    ];
+
+    const signature = MPC.constructSignature([signA, signB]);
+
+    return signature;
+  }
+
+  async recoverTSS(params: RecoverOptions): Promise<RecoveryInfo | OfflineVaultTxInfo> {
+    // this.validateRecoveryParams(params);
+    const isUnsignedSweep = getIsUnsignedSweep(params);
+
+    // Clean up whitespace from entered values
+    const userKey = params.userKey.replace(/\s/g, '');
+    const backupKey = params.backupKey.replace(/\s/g, '');
+
+    // Set new eth tx fees (using default config values from platform)
+    const gasLimit = new optionalDeps.ethUtil.BN(this.setGasLimit(params.gasLimit));
+    const gasPrice = params.eip1559
+      ? new optionalDeps.ethUtil.BN(params.eip1559.maxFeePerGas)
+      : new optionalDeps.ethUtil.BN(this.setGasPrice(params.gasPrice));
+
+    let backupKeyAddress;
+    let userKeyCombined;
+    let backupKeyCombined;
+
+    if (isUnsignedSweep) {
+      const backupKeyPair = new KeyPairLib({ pub: backupKey });
```

Comment:
I don't understand this line. You're assign backupKey to publicKey. Can backupKey be a publicKey? if so maybe rename backupKey to something more descriptive